homeassistant/components/fritzbox/cover.py

```diff
@@ -24,19 +24,21 @@ async def async_setup_entry(
     coordinator = get_coordinator(hass, entry.entry_id)
 
     @callback
-    def _add_entities() -> None:
+    def _add_entities(devices: set[str] | None = None) -> None:
         """Add devices."""
-        if not coordinator.new_devices:
+        if not devices and not coordinator.new_devices:
             return
+        if not devices:
+            devices = coordinator.new_devices
```

Comment:
Change order here.in all such functions 

if devices is None
  Devices = newdev
if not devices
  return

Reply:
I racked my brains for over an hour on how to make this as simple and elegant as possible, but I couldn't come up with this solution 🙈

Reply:
009eabf